bridge_adaptivity/api/tests/test_api.py

```diff
@@ -0,0 +1,155 @@
+
+
+class TestApiBase(APITestCase):
+    @classmethod
+    @patch('module.tasks.sync_collection_engines.apply_async')
+    def setUpTestData(cls, mock_apply_async):
+        cls.user = BridgeUser.objects.create(
+            username='test',
+            password='test',
+        )
+        cls.token = Token.objects.create(user=cls.user)
+        cls.collection = Collection.objects.create(name='col1', owner=cls.user)
+        cls.activity = Activity.objects.create(name='act1', collection=cls.collection, tags='test')
+
+    def setUp(self):
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token.key)
+
+
+class TestCollectionAPI(TestApiBase):
+    """
+    Test collection API endpoints.
+
+    CRUD requests are tested as well as request authorization.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        super(TestCollectionAPI, cls).setUpTestData()
+        cls.list_url = reverse('api:collection-list')
+        cls.detail_url = reverse('api:collection-detail', args=[cls.collection.id])
+
+    def test_get_collection(self):
+        response = self.client.get(self.list_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 1)
+
+    def test_api_collection_call_without_authorization(self):
+        self.client.credentials()
+        response = self.client.get(self.list_url)
+        self.assertEqual(response.status_code, 401)
+
+    @patch('module.tasks.sync_collection_engines.apply_async')
+    def test_update_collection(self, mock_apply_async):
+        new_name = 'new_collection'
+        data = {
+            'name': new_name,
+        }
+        response = self.client.patch(self.detail_url, data)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['name'], new_name)
+
+    @patch('module.tasks.sync_collection_engines.apply_async')
+    def test_create_collection(self, mock_apply_async):
+        name = 'test_collection'
+        data = {
+            'name': name,
+            "metadata": None,
+            "strict_forward": True,
+            "owner": self.user.id
+        }
+        response = self.client.post(self.list_url, data)
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.data['name'], name)
+        collections = self.client.get(self.list_url).data
+        self.assertEqual(len(collections), 2)
+
+    @patch('module.tasks.sync_collection_engines.apply_async')
+    def test_delete_collection(self, mock_apply_async):
+        name = 'test_collection'
+        data = {
+            'name': name,
+            "metadata": None,
+            "strict_forward": True,
+            "owner": self.user.id
+        }
+        response = self.client.post(self.list_url, data)
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.data['name'], name)
+        deleted = self.client.delete(self.list_url + '{}/'.format(response.data['id']))
```

Comment:
should be better to rewrite 
`"{}{}/".format(self.list_url, response.data['id'])`

Reply:
or build reverse ?

Reply:
`reverse` seems to be the best choice. 
done.